ceramic-kubo-rpc/src/http/dag.rs

```diff
@@ -107,6 +110,24 @@ where
                     )
                     .await?
                 }
+                (DAG_CBOR, DAG_CBOR) => {
```

Comment:
Is there a way to capture the combinations more generically?

Reply:
The`dag::put`function takes generic type parameters for the input and output codecs. This means that we need to enumerate which ones we support so the Rust compiler can compile each combination of codecs that we use. Therefore this match expression makes it very clear which combinations we support. 

We could change the design of the dag::put function to not take generic type parameters and instead take a dynamic value that is known to implements the codec trait. This would allow us to pass any combination of codecs at runtime. This comes at the cost of requiring dynamic dispatch to call the implementations of the codec functions. This code is not in a hot path so we would likely not notice the cost of dynamic dispatch in this case. However its simple enough to enumerate the finite set of codecs we support so for now the type parameter design made the most sense.